### PR TITLE
Run as arbitrary user

### LINKED ISSUE #142: Update Dockerfile/entrypoint.sh to not require anyuid scc.
The broker's `build/Dockerfile` should be updated to follow best practices employed by the apb execution environment base. If run without anyuid, the broker container will enter a crash loop due to the entrypoint.sh being unable to write to sed temp files in /etc/ansible-service-broker.

Container should be able to run as anyuid.

### PR DESCRIPTION
**Pull request by fabianvf**
Allow ansible-service-broker container to run with an arbitrary user id

Closes #142

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ prepare-build: install
 	cp "${GOPATH}"/bin/broker build/broker
 
 build: prepare-build
-	docker build ${BUILD_DIR} -t ${BROKER_IMAGE}:${TAG}
+	docker build -t ${BROKER_IMAGE}:${TAG} ${BUILD_DIR}
```
Comment (jmrodri) [inline Makefile]: :+1:

File: build/Dockerfile [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
 COPY entrypoint.sh ansible-service-broker /usr/bin/
 COPY broker /usr/bin/asbd
 
-USER ansibleservicebroker
+RUN chown -R ${USER_NAME}:0 /var/log/ansible-service-broker \
+ && chown -R ${USER_NAME}:0 /etc/ansible-service-broker \
+ && chmod -R g+rw /etc/ansible-service-broker /var/log/ansible-service-broker
+
+USER ${USER_UID}
+RUN sed "s@${USER_NAME}:x:${USER_UID}:@${USER_NAME}:x:\${USER_ID}:@g" /etc/passwd > ${BASE_DIR}/etc/passwd.template
```
Comment (jmrodri) [inline build/Dockerfile]: Ok this looks fine to me.

File: build/entrypoint.sh [hunk 3]
```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+USER_ID=$(id -u)
+if [ ${USER_UID} != ${USER_ID} ]; then
+  sed "s@${USER_NAME}:x:\${USER_ID}:@${USER_NAME}:x:${USER_ID}:@g" ${BASE_DIR}/etc/passwd.template > /etc/passwd
+fi
+
```
Comment (jmrodri) [inline build/entrypoint.sh]: :+1:

**Comment by fabianvf** [comment]
@jmontleon should I rebase this PR on your branch or does your branch implement similar changes?

File: Makefile (hunk 1 above)
Comment (jmontleon) [inline Makefile]: @fabianvf @eriknelson I think this needs to be rebased and changed under build-image now to:
`docker build -f ${BUILD_DIR}/Dockerfile-src -t ${BROKER_IMAGE}:${TAG} ${BUILD_DIR}`

**Comment by eriknelson** [comment]
@fabianvf Can you rebase this as of master and @jmontleon's comments? I will review after you ping me.

File: Makefile [hunk 4]
```diff
@@ -43,7 +43,7 @@ prepare-build-image: build
 	cp broker build/broker
 
 build-image: prepare-build-image
-	docker build -f ${BUILD_DIR}/Dockerfile-src ${BUILD_DIR} -t ${BROKER_IMAGE}:${TAG}
+	docker build -f ${BUILD_DIR}/Dockerfile-src -t ${BROKER_IMAGE}:${TAG} ${BUILD_DIR}
```
Comment (jmontleon) [inline Makefile]: Is the release-image build line also wrong?
docker build ${BUILD_DIR} -t ${BROKER_IMAGE}:${TAG}

Should it be:
docker build -t ${BROKER_IMAGE}:${TAG} ${BUILD_DIR} 

If so can that get fixed as well?

Comment (fabianvf) [inline Makefile]: Yeah, no problem. Don't know why but it seems to only not work for me, my docker only allows `docker build [OPTIONS] DIR`

**Comment by eriknelson** [comment]
Tested and working, spoke with @jmontleon and he's happy. ACK

**Pull request merged**